### DISCUSSION TITLE
multi-vendor, single shop front

### DISCUSSION THREAD

**Discussion by tlodge** [vendurehq/vendure#1700] [Q&A] [answered]
I'm working on a vendure site that allows multiple vendors to sell their products on a single shopfront.  I'm using channels so that  each user can manage their inventory, and I have an eventbus subscription that replicates a core set of facets when a new vendor is created. 

I'm now trying to work out the best way to handle orders when a shopper selects and buys items that are from different vendors.  I want each vendor to see an order in their admin interface that only shows that items in the order that were sold by them.  I think that the way to do this is to do the following (in a plugin):

- when an order is made, pull out the channelIds for each product variant (each product variant will be associated with the default channel and the channel of the vendor that created the product) 
- create a new order for each collection of items and associate them with the vendor's channel id - that way, each vendor will see a new order that only relates to their product.
- remove the orginal order, but create a lookup to link the original order id to the new orderids (so that the customer only needs a single order id).

Does this seem reasonable?  I'm not entirely sure if there might be some problems later down the line, such as what happens with order fulfilment; but I'm willing to deal with these assuming that what I've outlined seems like a good way to go.

Thanks in advance for any responses.

**Comment by michaelbromley** [comment, marked as answer]
Hi Tom,

The problem you are trying to solve here is actually the main big feature I'm planning for Vendure v2 - see #1329.

You plan sounds workable, my main question is regarding the step "remove the original order". I wouldn't recommend actual physical deletion of the row - there would be foreign key constraints that make this a bad idea. But you could perhaps mark it as deleted with a flag. 

There will no doubt be further issues & edge cases to deal with - the order process is by far the most complex part of Vendure, so it's very difficult even for me to foresee all the ramifications of changes to the flow. In any case I'd be interested to hear your feedback if you go down this route as it could be useful in informing the future design of first-class multi-vendor support.

**Reply by tlodge** [reply]
Thanks for this michael.  I've been digging into it and reached the same conclusion.  I'll continue working on it and will gladly feed back to let you know how I get on.
